api/envoy/extensions/upstreams/http/v3/http_protocol_options.proto

```diff
@@ -150,7 +150,11 @@ message HttpProtocolOptions {
     AutoHttpConfig auto_config = 5;
   }
 
-  // [#not-implemented-hide:]
+  // .. warning::
+  //   HTTP filters are not supported by default.
+  //   This warning will be removed as support moves beyond alpha.
+  //
   // Optional HTTP filters for the upstream filter chain.
+  // [#extension-category: envoy.filters.http.upstream]
```

Comment:
For my education, why did we decide to put the filter chain here vs. how this is done for downstream? Curious about the API symmetry/consistency. @envoyproxy/api-shepherds

Reply:
I see, so rather than having the set of upstream filters specified once in the HCM with config overrides set in the cluster, we just specify the set of upstream filters in every cluster?

Reply:
right now we set them in each cluster.  I think we want to have a default in the bootstrap config, and overrides in every cluster, but we can add that on later (and again I'm hoping you can pick up that part)

Reply:
I thought for downstream we had one filter chain, and per route filter configs. Did we add more enhancements when I wasn't looking?

Reply:
Ah yeah sorry I was confusing route config vs. filter chain config. I agree though that if we could support it or at least future proof the API for it it would be nice.

Reply:
Also I still have the question about upstream vs. downstream. Do we expect this to work for downstream also? Should this be part of the upstream options? I think there was another comment thread on this but I can't find it.

Reply:
this proto is in the namespace envoy.upstreams.http.http_protocol_options

It's the HTTP protocol options that go in cluster config for configuring if an upstream is doing explicit HTTP1/HTTP2/HTTP3, auto, etc.

I think this is the correct place to add http_filters which are a per-cluster upstream filter chain as it's http-specific config, cluster-specific config, and upstream only.

Reply:
Ah OK sorry I missed that. I though this was generic protocol options. SGTM.